iset.mm

```diff
@@ -149346,6 +149393,232 @@ the additive structure must be abelian (see ~ ringcom ), care must be
+        TUFUNVBVLURUNUSVIURUOWBTUFUGUAUGUHHUPUSGACDVSWAFULUNBUIIVRVQOABCEUJHVGV
+        IGBUIVRVGSVISVRSULUMUK $.
+    $}
+
+    ${
+      $d V x y $.
+      oppr1.2 $e |- .1. = ( 1r ` R ) $.
+      $( Multiplicative identity of an opposite ring.  (Contributed by Mario
+         Carneiro, 1-Dec-2014.) $)
+      oppr1g $p |- ( R e. V -> .1. = ( 1r ` O ) ) $=
+        ( vx vy wcel cfv cbs co wceq wa wral cio eqid eqeq1d anbi12d cvv c0g cv
+        cmgp cur cmulr crio opprmulg 3expa simpll simpr simplr syl3anc biancomd
+        ralbidva riotabidva df-riota 3eqtr3g cplusg opprex mgpex grpidvalg 3syl
+        opprbasg mgpbasg eqtrd eleq2d mgpplusgg oveqd raleqbidv iotabidv eqtr4d
+        syl 3eqtr4d ringidvalg 3eqtr4rd ) ADIZCUCJZUAJZAUCJZUAJZCUDJZBVPGUBZAKJ
+        ZIZWBHUBZCUEJZLZWEMZWEWBWFLZWEMZNZHWCOZNZGPZWDWBWEAUEJZLZWEMZWEWBWOLZWE
+        MZNZHWCOZNZGPZVRVTVPWLGWCUFXAGWCUFWNXCVPWLXAGWCVPWDNZWKWTHWCXDWEWCIZNZW
+        KWQWSXFWHWSWJWQXFWGWRWEVPWDXEWGWRMWCAWFWOWCCDWCWBWEWCQZWOQZEWFQZUGUHRXF
+        WIWPWEXFVPXEWDWIWPMVPWDXEUIXDXEUJVPWDXEUKWCAWFWOWCCDWCWEWBXGXHEXIUGULRS
+        UMUNUOWLGWCUPXAGWCUPUQVPVRWBVQKJZIZWBWEVQURJZLZWEMZWEWBXLLZWEMZNZHXJOZN
+        ZGPZWNVPCTIZVQTIVRXTMACDEUSZCVQTVQQZUTHXJXLGVQTVRXJQXLQVRQVAVBVPWMXSGVP
+        WDXKWLXRVPWCXJWBVPWCCKJZXJWCACDEXGVCVPYAYDXJMYBYDCVQTYCYDQVDVLVEZVFVPWK
+        XQHWCXJYEVPWHXNWJXPVPWGXMWEVPWFXLWBWEVPYAWFXLMYBCWFVQTYCXIVGVLZVHRVPWIX
+        OWEVPWFXLWEWBYFVHRSVISVJVKVPVTWBVSKJZIZWBWEVSURJZLZWEMZWEWBYILZWEMZNZHY
+        GOZNZGPZXCVPVSTIVTYQMAVSDVSQZUTHYGYIGVSTVTYGQYIQVTQVAVLVPXBYPGVPWDYHXAY
+        OVPWCYGWBWCAVSDYRXGVDZVFVPWTYNHWCYGYSVPWQYKWSYMVPWPYJWEVPWOYIWBWEAWOVSD
+        YRXHVGZVHRVPWRYLWEVPWOYIWEWBYTVHRSVISVJVKVMVPYAWAVRMYBCWAVQTYCWAQVNVLAB
+        VSDYRFVNVO $.
+    $}
+
+    ${
+      $d V x y $.
+      opprneg.2 $e |- N = ( invg ` R ) $.
+      $( The negative function in an opposite ring.  (Contributed by Mario
+         Carneiro, 5-Dec-2014.)  (Revised by Mario Carneiro, 2-Oct-2015.) $)
+      opprnegg $p |- ( R e. V -> N = ( invg ` O ) ) $=
+        ( vx vy wcel cbs cfv cv cplusg co c0g wceq crio cmpt eqid grpinvfvalg
+        cminusg opprbasg oppraddg oveqd oppr0g riotaeqbidv mpteq12dv cvv opprex
+        eqeq12d syl 3eqtr4d ) ADIZGAJKZHLZGLZAMKZNZAOKZPZHUNQZRGCJKZUOUPCMKZNZC
+        OKZPZHVBQZRZBCUAKZUMGUNVAVBVGUNACDEUNSZUBZUMUTVFHUNVBVKUMURVDUSVEUMUQVC
+        UOUPUQACDEUQSZUCUDACDUSEUSSZUEUJUFUGGHUNUQABDUSVJVLVMFTUMCUHIVIVHPACDEU
+        IGHVBVCCVIUHVEVBSVCSVESVISTUKUL $.
+    $}
+  $}
+
+  ${
+    $d x y B $.  $d x y N $.  $d x y R $.  $d x y X $.  $d x y Y $.
+    mulgass3.b $e |- B = ( Base ` R ) $.
+    mulgass3.m $e |- .x. = ( .g ` R ) $.
+    mulgass3.t $e |- .X. = ( .r ` R ) $.
+    $( An associative property between group multiple and ring multiplication.
+       (Contributed by Mario Carneiro, 14-Jun-2015.) $)
+    mulgass3 $p |- ( ( R e. Ring /\ ( N e. ZZ /\ X e. B /\ Y e. B ) ) ->
```

Comment:
In this theorem, there is no "opposite ring". It is, however, used in the proof. I would suggest that this is mentioned in the comment, e.g. "This theorem can be proved easily using opposite rings."

Reply:
I guess to me the opposite ring is sort of a housekeeping detail and the more fundamental part of the proof is that it builds on https://us.metamath.org/ileuni/mulgass2.html .

My preference, most of the time, is to let the formal proofs speak for themselves. I guess we could say something like "The proof uses an opposite ring to derive this from ~ mulgass2" but I'm having trouble coming up with a wording that thrills me.

Reply:
I simply wondered why we have a theorem without containing an opposite ring in the section about opposite rings, and I was just about to suggest to move it up (near ~mulgass2), but this is not possible because of the usage of opposite rings in the proof. But my remark is non-blocking.

Reply:
Yeah, I had a little bit of the reaction of wondering why it is in this section. Whether adding some kind of comment makes it any better is what gives me pause.

I guess I'll merge this without any comment change, probably tomorrow.